### PR TITLE
Add ByteArray assigment

### LINKED ISSUE #236: Support c# byte[] assignment operation
Here is an example:
```cs
var a = new byte[]{ 0x00, 0x01}
a[0] = 0x01
```

### PR DESCRIPTION
**Pull request by ShawnYun**
close #236 

1. Add ByteArray assigment.
2. Fix Byte[] to Buffer.

### CODE REVIEW

File: src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs [hunk 1]
```diff
@@ -1227,6 +1227,7 @@ private int ConvertNewArr(ILMethod method, OpCode src, NeoMethod to)
                         if (bLdLoc == false)//It means there's no initialization at all
                         {
                             this.ConvertPushDataArray(outbyte, src, to);
+                            Insert1(VM.OpCode.CONVERT, "", to, new byte[1] { (byte)VM.Types.StackItemType.Buffer });
```
Comment (Tommo-L) [inline src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs:1230]:
```suggestion
                            Insert1(VM.OpCode.CONVERT, "", to, new byte[] { (byte)VM.Types.StackItemType.Buffer });
```

Comment (Tommo-L) [reply src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs:1230]: Plz, keep the naming rules consistent.

Comment (ShawnYun) [reply src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs]: Done

**Comment by lightszero** [comment]
please fix unittest

**Comment by ShawnYun** [comment]
@lightszero Has fixed.

**Comment by ShawnYun** [comment]
The length of the function name cannot exceed 32.

**Pull request merged**